src/main/java/com/faforever/client/chat/UserFilterController.java

```diff
@@ -124,23 +134,42 @@ boolean isGameStatusMatch(ChatUserItemController chatUserItemController) {
   }
 
   public void onGameStatusPlaying(ActionEvent actionEvent) {
-    playerStatusFilter = PLAYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.playing"));
-    filterUsers();
+    setOrUnsetGameStatusFilter(PLAYING);
   }
 
   public void onGameStatusLobby(ActionEvent actionEvent) {
-    playerStatusFilter = LOBBYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.lobby"));
-    filterUsers();
+    setOrUnsetGameStatusFilter(LOBBYING);
   }
 
   public void onGameStatusNone(ActionEvent actionEvent) {
-    playerStatusFilter = IDLE;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.none"));
+    setOrUnsetGameStatusFilter(IDLE);
+  }
+
+  private void setOrUnsetGameStatusFilter(PlayerStatus newStatus) {
```

Comment:
"Set or unset" is not very intuitive to understand, a bit magical and thus error-prone.
I'd rather like to see the call like this:

```
if(((Toggle) actionEvent.getTarget()).isSelected()) {
  setGameStatusFilter(IDLE);
} else {
  setGameStatusFilter(null);
}
```

Yes, it's more code and less fancy, but very explicit and thus easy to understand, no magic involved.